composer/callbacks/checkpoint_saver.py

```diff
@@ -392,7 +393,7 @@ def _save_checkpoint(self, state: State, logger: Logger, log_level: LogLevel):
                     os.remove(symlink_name)
                 except FileNotFoundError:
                     pass
-                os.symlink(checkpoint_filepath, symlink_name)
+                os.symlink(Path(checkpoint_filepath).name, symlink_name)
```

Comment:
Hmm, would this work if the `checkpoint_filepath` contained a subfolder within the checkpoint folder?

E.g. checkpoint folder = `foo`
checkpoint_filename = `rank{rank}/epoch{epoch}_batch{batch}`
latest = `rank_{rank}`

Then would this write `foo/rank_0` to point to `foo/rank0/epoch0_batch1`, for example?

(Sorry, I don't really get how symlinks work, but want to make sure this covers the edge cases)

Reply:
Good catch -- fixed it to work in case `checkpoint_filename` is a path instead of just a filename